solidity/dashboard/src/components/Timer.jsx

```diff
@@ -1,8 +1,54 @@
-import React from "react"
+import React, { useEffect, useState } from "react"
 
-const Timer = ({ target }) => {
-  // TODO implement timer
-  return <>{target}</>
+const Timer = ({ days, hours, minutes, seconds }) => {
+  const [remainingDays, setRemainingDays] = useState(days || 0)
+  const [remainingHours, setRemainingHours] = useState(hours || 0)
+  const [remainingMinutes, setRemainingMinutes] = useState(minutes || 0)
+  const [remainingSeconds, setRemainingSeconds] = useState(seconds || 0)
+
+  const zeroPad = (num, places) => String(num).padStart(places, "0")
+
+  useEffect(() => {
+    const myInterval = setInterval(() => {
+      if (remainingSeconds > 0) {
+        setRemainingSeconds(remainingSeconds - 1)
+      }
+      if (remainingSeconds === 0) {
+        if (
+          remainingDays === 0 &&
+          remainingHours === 0 &&
+          remainingMinutes === 0
+        ) {
+          clearInterval(myInterval)
+        } else {
+          setRemainingSeconds(59)
+          if (remainingMinutes > 0) {
+            setRemainingMinutes(minutes - 1)
+          } else if (remainingMinutes === 0) {
+            setRemainingMinutes(59)
+            if (remainingHours > 0) {
```

Comment:
nested ifs... 🤯  out of idea now how we can do better, but let's keep 👀 on that.

Reply:
One way to get cleaner code would be to keep the remainingSeconds only and then when displaying we could format it with moment js. However, I don't think it will be efficient to do such thing every re-render (which means every sec). Thoughts?

Reply:
> However, I don't think it will be efficient to do such thing every re-render (which means every sec)

Agree. Let's keep it in mind, maybe someday we'll do it better 😆